src/compiler/OpenAPIYamlCompiler.ts

```diff
@@ -0,0 +1,164 @@
+     */
+    private static async executePromisesSequentially(promises: Array<() => Promise<any>>): Promise<any> {
+        for(const promise of promises) {
+            await promise();
+        }
+    }
+
+    /**
+     * Compiles OpenAPI yaml
+     */
+    public compile(): Promise<CompilationResult> {
+        console.log(`⟲ [${this.pluginName}] starting OpenAPI YAML compilation...`);
+        const start = new Date().getTime();
+        return this.buildPluginTypes(this.pluginName)
+            .then(() => {
+                let end = new Date().getTime();
+                console.log(GREEN_CHECK, `[${this.pluginName}] OpenAPI YAML finished (${formatDuration(end - start)})`);
+                return CompilationResult.CHANGED;
+            })
+            .catch((err) => {
+                return this.cleanup(this.pluginName).finally(() => {
+                    console.error(cerr`${err}`);
+                    throw Error(`[${this.pluginName}] Failed to write OpenAPI YAML output`);
+                });
+            });
+    }
+
+    /**
+     * Removes generated data
+     * @param plugin Plugin name
+     * @private
+     */
+    private cleanup(plugin: string): Promise<any> {
+        return Promise.all([
+            this.removeGeneratedOpenAPIFiles(plugin),
+            this.removePluginDist(plugin)
+        ]);
+    }
+
+    /**
+     *  Builds types for a provided plugin. First it generates types, than it
+     *  copies types in plugin/assets/ts/api folder, and at the end it cleans
+     *  the distribution folder.
+     * @param plugin Provided plugin
+     */
+    private buildPluginTypes(plugin: string): Promise<any> {
+        return OpenAPIYamlCompiler.executePromisesSequentially([
+            this.generatePluginTypes.bind(this, plugin),
+            this.copyPluginTypes.bind(this, plugin),
+            this.cleanup.bind(this, plugin)
+        ]);
+    }
+
+    /**
+     * Generates types from the API specification and place them in plugin/api/dist/openapi
+     * folder. For generation it uses openapi-generator-cli.
+     * @param plugin Provided plugin for which types should be generated
+     */
+    private generatePluginTypes(plugin: string): Promise<any> {
+        return new Promise((resolve) => {
+            const pluginPath = this.getPluginPath(plugin);
+            const cli = path.resolve(OpenAPIYamlCompiler.getNodeModulesBinPath(), 'openapi-generator-cli');
+            const yaml = path.resolve(`${pluginPath}/api/API.yaml`);
+            const dist = path.resolve(`${pluginPath}/api/dist/openapi`);
+            const vmArgs = 'version-manager set 5.0.0'.split(' ');
+            const vmRes = spawn.sync(cli, vmArgs, {
+                stdio: ['pipe', 'pipe', process.stderr]
+            });
+            const genArgs = `generate -i ${yaml} -g typescript-angular -o ${dist} --additional-properties=ngVersion=6.1.7,npmName=restClient,supportsES6=true,npmVersion=6.9.0,withInterfaces=true`.split(' ');
```

Comment:
could it make sense to make use of openapitools.json instead of hardcoding the options?

Reply:
okay I think the options will hardly change much over time

Reply:
Yeah. It will probably never change. And with openapitools.json it was more difficult to set glob and output.